### PR TITLE
[RISCV] Assign separate PseudoVSHA2MS_VV opcodes for each SEW

### PR DESCRIPTION
**Pull request by mshockwave**
The vsha2ms.vv from Zvknh[ab] currently supports both SEW=32 and SEW=64. It might have different performance characteristics depending on the SEW on some processors. This patch splits these two different SEWs into their own VPsuedo opcodes and scheduling classes.

This is effectively a NFC change.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td [hunk 1]
```diff
@@ -4454,21 +4454,22 @@ class VPatTernaryNoMaskWithPolicy<string intrinsic,
...
                                   ValueType op1_type,
                                   ValueType op2_type,
-                                  int sew,
+                                  int log2sew,
                                   LMULInfo vlmul,
                                   VReg result_reg_class,
                                   RegisterClass op1_reg_class,
-                                  DAGOperand op2_kind> :
+                                  DAGOperand op2_kind,
+                                  bit sew_aware = false> :
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td:4462]: Why sew_aware instead of isSEWAware like everywhere else?

Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td]: Fixed.

File: llvm/lib/Target/RISCV/RISCVInstrInfoZvk.td [hunk 2]
```diff
@@ -564,7 +566,9 @@ let Predicates = [HasStdExtZvkned] in {
 let Predicates = [HasStdExtZvknhaOrZvknhb] in {
   defm PseudoVSHA2CH : VPseudoVSHA2CH;
   defm PseudoVSHA2CL : VPseudoVSHA2CL;
-  defm PseudoVSHA2MS : VPseudoVSHA2MS;
+  defm PseudoVSHA2MS : VPseudoVSHA2MS<sew=32>;
+  let Predicates = [HasStdExtZvknhb] in
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoZvk.td:570]: Whic `let` wins here. This one or the one on line 566?

Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoZvk.td:570]: this one (line 570) wins.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**